content/2017.12.kinto-at-mozilla.rst

```diff
@@ -0,0 +1,148 @@
+
+Resist fingerprinting with fonts whitelist
+------------------------------------------
+
+`The idea <https://bugzilla.mozilla.org/show_bug.cgi?id=1336208>`_ is to deliver some fonts to avoid downloading them while browsing. The client globally works like the Android assets.
+
+Possible evolutions
+-------------------
+
+- There is certainly going to be more types of settings for new use-cases
+- The stack currently uses `kinto-ldap` to authenticate users. We'll switch to OpenID Connect now that the company uses Auth0 everywhere.
+- The `Push team` is working on a solution to replace frequent polling with a push (at that scale nothing is trivial).
+- The Sync team is working on a new data store for profile data. We'll probably replace the current SQlite adapter.
+
+
+WebExtensions storage.sync API
+==============================
+
+The tough one!
+
+Firefox is now compatible with the Web Extensions APIs, that allow addons developers to share a common code-base between different browsers. One of these APIs allows addons to store and sync some data (eg. user-preferences).
+In our implementation, the client encrypts the data before sending it to our servers to guarantee privacy. It might not be the case of Chrome by the way. It uses `kinto.js` and leverages the Firefox Accounts keyring.
+
+This stack allowed us to really stress Kinto. When popular extensions like Adblock switched to Web Extensions, we had database congestions and lots of 5XX errors. We should have been a lot more thorough with load testing, and hence discovered race conditions and scaling bugs the hard way. Most of them are fixed now though.
+
+The quota plugin restricts the amount of objects allowed to be stored. It relies on two Memcache instances as cache backends, mainly used for caching authentication.
+
+Instance: https://webextensions.settings.services.mozilla.com
+Production date: July 2017
+Average requests/sec: 1100 (70K/min)
+Data size: XXX millions buckets, XXX collections, XXX records
+
+Possible evolutions
+-------------------
+
+- We still have some `TOCTOU issues <https://en.wikipedia.org/wiki/Time_of_check_to_time_of_use>`_ that we Would like to tackle.
+- At some point, we might have to consider some sharding strategy too.
+
+
+Firefox Test Pilot
+==================
+
+Authentication is provided by Firefox Accounts and assured by `kinto-fxa`.
+
+The main challenge here was to reuse the same stack for different Test Pilot extensions, while being able to maintain data isolation and client side encryption.
+
+Each test pilot addon has its own scope, so that an extension has no permission to read the server data produced by another one (*using a kinto-fxa trick that adds prefixes to user ids*). Plus, the client side code fetches a different encryption key per extension, which means an extension has no way to decrypt the local data produced by another one.
```

Comment:
You use italics here in a way that I've seen other non-native speakers use them and found strange. It almost seems like you're using them not to add emphasis (which is what I would expect) but to make them even more parenthetical, like a comment in source code. I probably wouldn't use italics here, but I hesitate to tell you to take them out.. they seem like an expression of a certain kind of voice.